src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.cc

```diff
@@ -295,14 +342,52 @@ OrphanablePtr<LoadBalancingPolicy> XdsOverrideHostLb::CreateChildPolicyLocked(
   return lb_policy;
 }
 
+RefCountedPtr<SubchannelInterface> XdsOverrideHostLb::LookupSubchannelByAddress(
+    absl::string_view address) {
+  absl::MutexLock lock(&subchannel_by_address_map_mu_);
+  std::string key{address};
+  auto subchannel_record = subchannel_by_address_map_.find(key);
+  if (subchannel_record == subchannel_by_address_map_.end()) {
+    if (GRPC_TRACE_FLAG_ENABLED(grpc_lb_xds_override_host_trace)) {
+      gpr_log(
+          GPR_INFO,
+          "[xds_override_host_lb %p] Subchannel for address %s was not found",
+          this, key.c_str());
+    }
+    return nullptr;
+  }
+  if (GRPC_TRACE_FLAG_ENABLED(grpc_lb_xds_override_host_trace)) {
+    gpr_log(GPR_INFO,
+            "[xds_override_host_lb %p] Subchannel for address %s was found",
+            this, key.c_str());
+  }
+  return subchannel_record->second.subchannel;
+}
+
+void XdsOverrideHostLb::RegisterSubchannel(
+    const ServerAddress& address,
+    RefCountedPtr<SubchannelInterface> subchannel) {
+  absl::MutexLock lock(&subchannel_by_address_map_mu_);
+  auto key = MakeKeyForAddress(address);
+  if (GRPC_TRACE_FLAG_ENABLED(grpc_lb_xds_override_host_trace)) {
+    gpr_log(GPR_INFO,
+            "[xds_override_host_lb %p] Added subchannel with address %s", this,
+            key.c_str());
+  }
+  subchannel_by_address_map_[key] = {subchannel};
+}
+
 //
 // XdsOverrideHostLb::Helper
 //
 
 RefCountedPtr<SubchannelInterface> XdsOverrideHostLb::Helper::CreateSubchannel(
     ServerAddress address, const ChannelArgs& args) {
-  return xds_override_host_policy_->channel_control_helper()->CreateSubchannel(
-      address, args);
+  auto subchannel =
+      xds_override_host_policy_->channel_control_helper()->CreateSubchannel(
+          address, args);
+  xds_override_host_policy_->RegisterSubchannel(address, subchannel);
```

Comment:
This handles adding a subchannel to the map, but it doesn't look like we're handling *removing* a subchannel from the map anywhere.  If the child policy unrefs a subchannel, we should not continue to hold a ref to it in the map, because that could potentially hold open a connection that is not actually needed.

I don't think we can do this if we hold a ref to the subchannel in our map, because then we have no way to know when the child policy has unreffed the subchannel, since its ref-count will never go to zero.  Instead, I think we need a structure here similar to that of the `outlier_detection` policy:

- The `AddressMapEntry` struct should become a ref-counted class.  (This is analogous to the `SubchannelState` class in the `outlier_detection` policy.)
- When a subchannel is created, we wrap it in our own subchannel object, which will hold a ref to the relevant `AddressMapEntry`.
- The `AddressMapEntry::subchannel` field should should be a raw pointer instead of a `RefCountedPtr<>` -- it should not hold ownership of the subchannel, since that will prevent the subchannel from being destroyed by the child policy.  Instead, when the subchannel wrapper is unreffed by the child policy, its dtor can check whether `AddressMapEntry::subchannel` is set to point to `this`, and if so, it will reset it to null.

I think this approach also points to the right way to track the subchannel connectivity state: the subchannel wrapper can intercept the watcher started by the child policy, and it can update the state in the `AddressMapEntry` before passing it on to the child policy.

Reply:
Done.